flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperatorMigrationTest.java

```diff
@@ -149,9 +169,16 @@ public void testRestoreSessionWindowsWithCountTrigger() throws Exception {
 				new KeyedOneInputStreamOperatorTestHarness<>(operator, new TupleKeySelector(), BasicTypeInfo.STRING_TYPE_INFO);
 
 		testHarness.setup();
-		testHarness.initializeState(
-				OperatorSnapshotUtil.readStateHandle(
-						OperatorSnapshotUtil.getResourceFilename("win-op-migration-test-session-with-stateful-trigger-flink1.2-snapshot")));
+
+		String savepointFile = "win-op-migration-test-session-with-stateful-trigger-flink" + testMigrateVersion + "-snapshot";
+		if (testMigrateVersion.equals("1.1")) {
```

Comment:
I think you could refactor this piece of code into a method, something like:
```
restoreFromSnapshot(TestHarness, SnapshotPath, MigrationVersion)
```

What do you think?